Core/GDCore/Events/CodeGeneration/EventsCodeGenerator.cpp

```diff
@@ -68,6 +68,16 @@ gd::String EventsCodeGenerator::GenerateRelationalOperatorCall(
     }
   }
 
+  // Check type of Relational Operator
+  if (relationalOperator == "startsWith") {
+    return rhs + " != '' && " + callStartString + "(" + argumentsStr + ").match(new RegExp('^' + " + rhs + ")) !== null";
+  } else if (relationalOperator == "endsWith") {
+    return rhs + " != '' && " + callStartString + "(" + argumentsStr + ").match(new RegExp(" + rhs + " + '$')) !== null";
+  } else if (relationalOperator == "contains") {
+    return rhs + " != '' && " + callStartString + "(" + argumentsStr + ").match(new RegExp(" + rhs + ")) !== null";
+  }

```

Comment:
I guess it can be simplified now.
```suggestion
  if (relationalOperator == "startsWith") {
    return rhs + " != '' && " + callStartString + "(" + argumentsStr + ").startsWith(" + rhs + ")";
  } else if (relationalOperator == "endsWith") {
    return rhs + " != '' && " + callStartString + "(" + argumentsStr + ").endsWith(" + rhs + ")";
  } else if (relationalOperator == "contains") {
    return rhs + " != '' && " + callStartString + "(" + argumentsStr + ").contains(" + rhs + ")";
  }
```

Reply:
Almost, but I guess there is a need for:
- more parentheses (around rhs, around the whole &&) to be sure nothing take precedence over this. 
- to *not* evaluate `rhs` twice, this could have at best performance implication or at worst side effect because running some expressions twice (though expressions should not have side effects... but still).

I think this means we have to offer a `gdjs.evtTools.stringStartsWith(lhs, rhs), that does lhs.startsWith(rhs)`, or return false if rhs is empty.
Also same for `contains` which I think does not exist on JavaScript strings.

Reply:
> Also same for `contains` which I think does not exist on JavaScript strings.

Indeed, it's `includes`.

Reply:
I think the initial check could even be removed.
When 2 empty strings are compared, we return `true` for `=` so it would be odd that `false` is returned for `startsWith`, `endsWith` and `contains` (btw should it be renamed `includes`?).